### PR TITLE
Scripts refactoring

### PR DESCRIPTION
**Pull request by meunierfrederic**
Thank you for your work on geo data. It's really useful!
Here a pull request which I hope will help you to maintain this component. 

I Cant figure out why XXXNestedFiles are use by CombineSplitFilesToMakeALargeDataset. It should use only the splitted files edited by contributors, shouldn't it?

What I've done: 

- apply splitLargeDatasetFiles-CSC.ts to populate data folder with uptodate splitted files
- refactor and apply combineSplitFilesToMakeALargeDatasetFiles-CSC.ts  in order to take only splitted files (not XXXNestedFiles) and write result in assets/allXXXNew.json files
- apply verificationCombine.ts to check that original assets files and allXXXNew files are equivalent

Notes: 
- a couple of splitted files were not sync with original assets files (concerning city "AU-WA-Woodlands")
- I've written some sort utilities for Country, State (by country and then State isoCodes) and City (by Country, State isoCodes and then by City names)
- my vscode doesn't infer that assets/cities.json contains an Array of object; I've added a cast when necessary
- there is an "inconsistent" configuration of prettier (indent=4 spaces) and the JSON.stringify (3 spaces)  for splitted files... Don't know how to fix that easily...
- this PR includes PR #129 

Thank you and enjoy reviewing this PR. Don't hesitate to contact me if you need any additional info

Frédéric

### CODE REVIEW

**Comment by harpreetkhalsagtbit** [comment]
Thanks @meunierfrederic for the contribution, I couldn't get a chance to properly check this but I did it yesterday.

1. `XXXNestedFiles` these are for future ref if any
2. I will check your refactoring and notes.

Thanks.

**Comment by harpreetkhalsagtbit** [comment]
@meunierfrederic 
Please help me understand, How can `gz` help us here?

File: src/scripts/combineSplitFilesToMakeALargeDatasetFiles-CSC.ts [hunk 1]
```diff
@@ -22,60 +28,119 @@ main.map((item: any, index: number) => {
...
+		}, allStatesGeo);
+
+		[...allStatesLite.keys()].forEach((key) => {
+			assert(allStatesGeo.get(key));
+			const combine: IState = { ...allStatesLite.get(key)!, ...allStatesGeo.get(key)! };
+			allStates.push(combine);
+		});
+	} catch (error) {
+		process.stdout.write(` ${error.message}\n`);
+	} finally {
```
Comment (harpreetkhalsagtbit) [inline src/scripts/combineSplitFilesToMakeALargeDatasetFiles-CSC.ts:69]: getting typescript error - Error is unknown

```js
		if (error instanceof Error) {
			// ✅ TypeScript knows err is Error
			console.log(error.message);
			process.stdout.write(` ${error.message}\n`);
		} else {
			console.log('Unexpected error', error);
		}

```

File: src/scripts/combineSplitFilesToMakeALargeDatasetFiles-CSC.ts [hunk 2]
```diff
@@ -22,60 +28,119 @@ main.map((item: any, index: number) => {
...
+	const allStatesLite = new Map<string, IState>();
+	const allStatesGeo = new Map<string, IState>();
+	try {
+		const { countryPath } = countryMetas[country.isoCode]; // `${country.name.replace(/\W/g, '_')}-${country.isoCode}`;
+		(JSON.parse(
+			fs.readFileSync(path.join(__dirname, PATH_TO_DATA_FOLDER, countryPath, 'allStates.lite.json'), 'utf-8'),
+		) as IState[]).reduce((accumulator, item) => {
+			accumulator.set(item.isoCode, item);
+			return accumulator;
+		}, allStatesLite);
```
Comment (harpreetkhalsagtbit) [inline src/scripts/combineSplitFilesToMakeALargeDatasetFiles-CSC.ts:53]: prettier issue in my vscode

Comment (harpreetkhalsagtbit) [reply src/scripts/combineSplitFilesToMakeALargeDatasetFiles-CSC.ts:53]: Screen Shot 2023-04-02 at 6 59 04 PM

**Review by harpreetkhalsagtbit — CHANGES_REQUESTED** [review]

**Comment by harpreetkhalsagtbit** [comment]
@meunierfrederic this is working perfectly. I just tested it with a state, city change and it worked fine. 
Thanks a lot!

**Comment by meunierfrederic** [comment]
> @meunierfrederic Please help me understand, How can `gz` help us here?

Since we deploy your component in some AWS Lambda we have some limitation for the unpacked bundle. And the country-state-city unpacked size is about 50M. When we deploy our fork of country-state-city with gz resources, the unpacked size is about 5M. The three gz files have been commited by error...

Anyway, I think it would be a good idea to distribute country-state-city with zipped resources (see Issue #93)

File: src/scripts/combineSplitFilesToMakeALargeDatasetFiles-CSC.ts (hunk 2 above)
Comment (meunierfrederic) [reply src/scripts/combineSplitFilesToMakeALargeDatasetFiles-CSC.ts:53]: Can you please tell me what prettier plugin you use? I'll use the same next time.

Comment (meunierfrederic) [reply src/scripts/combineSplitFilesToMakeALargeDatasetFiles-CSC.ts:53]: applying prettier once again fix the issue :-)

File: src/scripts/combineSplitFilesToMakeALargeDatasetFiles-CSC.ts (hunk 1 above)
Comment (meunierfrederic) [reply src/scripts/combineSplitFilesToMakeALargeDatasetFiles-CSC.ts:69]: can't reproduce this one... My vscode is using currently typescript 4.9.4, as configured in package.json...
image

**Comment by harpreetkhalsagtbit** [comment]
> > @meunierfrederic Please help me understand, How can `gz` help us here?
> 
> Since we deploy your component in some AWS Lambda we have some limitation for the unpacked bundle. And the country-state-city unpacked size is about 50M. When we deploy our fork of country-state-city with gz resources, the unpacked size is about 5M. The three gz files have been commited by error... I've removed them.
> 
> Anyway, I think it would be a good idea to distribute country-state-city with zipped resources (see Issue #93)

Yes, thats why I want to know, how it can help us to resolve that issue. How can we use `.gz` file in the code directly?

**Comment by meunierfrederic** [comment]
> > > @meunierfrederic Please help me understand, How can `gz` help us here?
> > 
> > 
> > Since we deploy your component in some AWS Lambda we have some limitation for the unpacked bundle. And the country-state-city unpacked size is about 50M. When we deploy our fork of country-state-city with gz resources, the unpacked size is about 5M. The three gz files have been commited by error... I've removed them.
> > Anyway, I think it would be a good idea to distribute country-state-city with zipped resources (see Issue #93)
> 
> Yes, thats why I want to know, how it can help us to resolve that issue. How can we use `.gz` file in the code directly?

If you like, I can work on it and create another PR on this topic.

**Review by harpreetkhalsagtbit — APPROVED** [review]

**Comment by harpreetkhalsagtbit** [comment]
> > > > @meunierfrederic Please help me understand, How can `gz` help us here?
> > > 
> > > 
> > > Since we deploy your component in some AWS Lambda we have some limitation for the unpacked bundle. And the country-state-city unpacked size is about 50M. When we deploy our fork of country-state-city with gz resources, the unpacked size is about 5M. The three gz files have been commited by error... I've removed them.
> > > Anyway, I think it would be a good idea to distribute country-state-city with zipped resources (see Issue #93)
> > 
> > 
> > Yes, thats why I want to know, how it can help us to resolve that issue. How can we use `.gz` file in the code directly?
> 
> If you like, I can work on it and create another PR on this topic.

Yes please. It would be great!

**Pull request merged**